bindings/pyroot/pythonizations/test/tfile_context_manager.py

```diff
@@ -0,0 +1,44 @@
+import os
+import unittest
+
+import ROOT
+
+from ROOT import TFile
+
+
+class TFileContextManager(unittest.TestCase):
+    """
+    Test of TFile used as context manager
+    """
+
+    NBINS = 100
+    XMIN = 0
+    XMAX = 100
+
+    def test_write_histogram(self):
+        """
+        Check that an histogram is correctly written to a file within a context
+        manager.
+        """
+        filename = "TFileContextManager_test_write_histogram.root"
+        with TFile(filename, "recreate") as outfile:
+            hout = ROOT.TH1F("h", "h", self.NBINS, self.XMIN, self.XMAX)
+            outfile.WriteObject(hout, "histo")
+
+        # Need to understand better what to check here. For now simple sanity checks
+        self.assertTrue(bool(outfile))  # The TFile object is still there
+        self.assertFalse(bool(outfile.Get("histo")))  # The histogram can't be retrieved from the same TFile
+        self.assertFalse(bool(hout))  # The `hout` variable points to a CPyCppyy_NoneType
```

Comment:
Here you are relying on the fact that `bool(None)` is `False` (that's what will be internally evaluated). Is this what you want?

Reply:
I think it's enough that you test that the file is closed, that's the sole action you do in your exit function, closing the file. The rest is behaviour that is derived from closing the file, which you don't strictly need to test here.